src/ui/run-history-ui-handler.ts

```diff
@@ -0,0 +1,317 @@
+      if (button === Button.ACTION) {
+        const cursor = this.cursor + this.scrollCursor;
+        if (this.runs[cursor]) {
+          this.scene.ui.setOverlayMode(Mode.RUN_INFO, this.runs[cursor].entryData, true);
+        } else {
+          return false;
+        }
+        success = true;
+        return success;
+      } else {
+        this.runSelectCallback = null;
+        success = true;
+        this.scene.ui.revertMode();
+      }
+    } else {
+      switch (button) {
+      case Button.UP:
+        if (this.cursor) {
+          success = this.setCursor(this.cursor - 1);
+        } else if (this.scrollCursor) {
+          success = this.setScrollCursor(this.scrollCursor - 1);
+        }
+        break;
+      case Button.DOWN:
+        if (this.cursor < 2) {
+          success = this.setCursor(this.cursor + 1);
+        } else if (this.scrollCursor < this.runs.length - 3) {
+          success = this.setScrollCursor(this.scrollCursor + 1);
+        }
+        break;
+      }
+    }
+
+    if (success) {
+      ui.playSelect();
+    } else if (error) {
+      ui.playError();
+    }
+    return success || error;
+  }
+
+
+  async populateRuns(scene: BattleScene) {
+    const response = await this.scene.gameData.getRunHistoryData(this.scene);
+    const timestamps = Object.keys(response);
+    if (timestamps.length === 0) {
+      return;
+    }
+    const timestampsNo = timestamps.map(Number);
+    if (timestamps.length > 1) {
+      timestampsNo.sort((a, b) => b - a);
+    }
+    const entryCount = timestamps.length;
+    for (let s = 0; s < entryCount; s++) {
+      const entry = new RunEntryContainer(this.scene, response[timestampsNo[s]], s);
+      this.scene.add.existing(entry);
+      this.runsContainer.add(entry);
+      this.runs.push(entry);
+    }
+  }
+
+  setCursor(cursor: number): boolean {
+    const changed = super.setCursor(cursor);
+
+    if (!this.cursorObj) {
+      this.cursorObj = this.scene.add.nineslice(0, 0, "select_cursor_highlight_thick", undefined, 308, 56, 6, 6, 6, 6);
+      this.cursorObj.setOrigin(0, 0);
+      this.runsContainer.add(this.cursorObj);
+    }
+    this.cursorObj.setPosition(-2, -2 + (cursor + this.scrollCursor) * 56);
+    return changed;
+  }
+
+  setScrollCursor(scrollCursor: number): boolean {
+    const changed = scrollCursor !== this.scrollCursor;
+
+    if (changed) {
+      this.scrollCursor = scrollCursor;
+      this.setCursor(this.cursor);
+      this.scene.tweens.add({
+        targets: this.runsContainer,
+        y: this.runContainerInitialY - 56 * scrollCursor,
+        duration: Utils.fixedInt(325),
+        ease: "Sine.easeInOut"
+      });
+    }
+
+    return changed;
+  }
+
+  clear() {
+    super.clear();
+    this.runSelectContainer.setVisible(false);
+    this.eraseCursor();
+    this.runSelectCallback = null;
+    this.clearruns();
+  }
+
+  eraseCursor() {
+    if (this.cursorObj) {
+      this.cursorObj.destroy();
+    }
+    this.cursorObj = null;
+  }
```

Comment:
NIT: Keep naming consistent

```suggestion
  destroyCursor() {
    if (this.cursorObj) {
      this.cursorObj.destroy();
    }
    this.cursorObj = null;
  }
```